### PR TITLE
electron 8 + spellcheck

### PR DESCRIPTION
**Pull request by chrisnojima**
Adds electron 8 plus spell checker

- [x] pref to disable checking
- [x] only main window for now, seems fine
- [x] cleanup advanced settings panel
- [x] move download url (cc: @maxtaco )
- [x] os x
- [ ] linux (cc: @heronhaye )
- [ ] windows (cc: @xgess )

### CODE REVIEW

**Comment by heronhaye** [comment]
I got a report from @eli-schwartz that the systray icon disappears on Linux on Electron 8

File: shared/common-adapters/box.desktop.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import './box.css'
 
 export class Box extends React.PureComponent<any> {
   render() {
-    const {forwardedRef, ...rest} = this.props
+    const {forwardedRef, onLayout, ...rest} = this.props
```
Comment (chrisnojima) [inline shared/common-adapters/box.desktop.tsx]: shouldn't be passed down

**Comment by eli-schwartz** [comment]
Specifically w.r.t. the systray icon, the issue is https://bugs.archlinux.org/task/65483 and therefore more generally, "electron 8 removes support for system tray icons other than StatusNotifier/AppIndicator". So it will depend on what your Desktop Environment or Window Manager is.

**Comment by chrisnojima** [comment]
maybe related: https://github.com/electron/electron/issues/21798

**Pull request merged**